src/map/storage.c

```diff
@@ -465,30 +504,34 @@ int guild_storage_additem(struct map_session_data* sd, struct guild_storage* sto
 		return 1;
 	}
 
-	if(itemdb->isstackable2(data)){ //Stackable
-		for(i=0;i<MAX_GUILD_STORAGE;i++){
-			if(compare_item(&stor->items[i], item_data)) {
-				if( amount > MAX_AMOUNT - stor->items[i].amount || ( data->stack.guildstorage && amount > data->stack.amount - stor->items[i].amount ) )
+	if (itemdb->isstackable2(data)) { // Stackable
+		for (i = 0; i < sd->guild->max_storage; i++) {
```

Comment:
This should loop on `i < stor->items.capacity` and `continue` if `stor->items.data[i].nameid <= 0`.